lib/SPIRV/SPIRVReader.cpp

```diff
@@ -685,49 +689,145 @@ void SPIRVToLLVM::setLLVMLoopMetadata(const LoopInstType *LM, Instruction *BI) {
                           LoopControlParameters[++NumParam])));
         break;
       }
-      default:
+      case DependencyArrayINTEL: {
+        // Collect array variable <-> safelen information
+        std::map<Value *, unsigned> ArraySflnMap;
+        unsigned NumOperandPairs = LoopControlParameters[++NumParam];
+        unsigned OperandsEndIndex = NumParam + NumOperandPairs * 2;
+        SPIRVModule *M = LM->getModule();
+        while (NumParam < OperandsEndIndex) {
+          SPIRVId ArraySPIRVId = LoopControlParameters[++NumParam];
+          Value *ArrayVar = ValueMap[M->getValue(ArraySPIRVId)];
+          unsigned Safelen = LoopControlParameters[++NumParam];
+          ArraySflnMap.emplace(std::make_pair(ArrayVar, Safelen));
+        }
+
+        // A single run over the loop to retrieve all GetElementPtr instructions
+        // that access relevant array variables
+        std::map<Value *, std::vector<GetElementPtrInst *>> ArrayGEPMap;
+        for (auto &BB : LoopObj->blocks())
+          for (Instruction &I : *BB) {
+            auto *GEP = dyn_cast<GetElementPtrInst>(&I);
```

Comment:
```suggestion
            if (auto *GEP = dyn_cast<GetElementPtrInst>(&I)) {
```

Reply:
The variable is re-used below, so this isn't going to work out if we're looking to retain the early `continue`.